OpenRA.Mods.Common/Traits/World/ShroudRenderer.cs

```diff
@@ -268,10 +268,10 @@ void WorldOnRenderPlayerChanged(Player player)
 			cellsDirty.Clear(true);
 			var tl = new PPos(0, 0);
 			var br = new PPos(map.MapSize.X - 1, map.MapSize.Y - 1);
-			UpdateShroud(new ProjectedCellRegion(map, tl, br));
+			UpdateShroud(new ProjectedCellRegion(map, tl, br).ToArray());
 		}
 
-		void UpdateShroud(ProjectedCellRegion region)
+		void UpdateShroud(PPos[] region)
```

Comment:
Would changing this to `IEnumerable<PPos>` regress performance?

If it doesn't then this would be the simplest way to address my last question - this method would be compatible with both `map.ProjectedCells` and `ProjectedCellRegion`.

Reply:
No, it won't. Will change.